### PR TITLE
Add issue template for code synchronization

### PR DESCRIPTION
**Pull request by israel-hdez**
#### Motivation

This adds a new GitHub issue template which covers synchronization of all ModelMesh related ODH repositories forks with upstream repositories.

Related to https://github.com/opendatahub-io/modelmesh-serving/issues/188

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/upstream-sync.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    echo "$TARGET_REPO_NAME: Clean"
+  else
+    echo "$TARGET_REPO_NAME: Conflict"
+  fi
+done
+```
+
+When the script finishes, you will see a sumary about merge status. You will need to manually fix any conflicts and finish the merge commiting the changes `git commit -s -m "Sync upstream ${RELEASE_BRANCH_NAME}"`
+
+Once all repositories are synced, you can push changes to your own fork and create PRs for all repositories, except `modelmesh-serving`.
```
Comment (spolti) [inline .github/ISSUE_TEMPLATE/upstream-sync.md:65]: As we discussed it earlier, can you please rephrase this part?

Comment (israel-hdez) [reply .github/ISSUE_TEMPLATE/upstream-sync.md]: @spolti I just updated the PR, FWIW.
Not sure if we still want to merge this.

Comment (spolti) [reply .github/ISSUE_TEMPLATE/upstream-sync.md]: Let's move to another place, it is indeed good content to serve as a doc.

Comment (israel-hdez) [reply .github/ISSUE_TEMPLATE/upstream-sync.md]: I just moved to `opendatahub/docs` hoping it is better.

**Review by spolti — APPROVED** [review]
/lgtm

**Pull request merged**